### DISCUSSION TITLE
Does infinispan guarantee eventual consistency in Replicated cache?

### DISCUSSION THREAD

**Discussion by nvinhphuc** [infinispan/infinispan#11166] [Q&A] [answered]
Our use case need in-memory replicated cache that is eventual consistent in all cases. In Hazelcast doc, Hazelcast **does not** guarantee eventual consistency for replicated map for some edge cases. We would like to know if Infinispan does guarantee this. If yes, can you explain in detail how Infinispan can handle this so I can see if it fits our use case? Thank you.

**Comment by tristantarrant** [comment, marked as answer]
No it does not. We've had a plan to implement it, but it never got enough priority. See https://issues.redhat.com/browse/ISPN-999

**Reply by nvinhphuc** [reply]
Thanks @tristantarrant , I see that the ticket suggest that Infinispan only support strong consistency, which might be the thing that we need. I would like to investigate some further questions:
- Can we only read in local cache and never allow to read in database?
- Do write operations block read operations?

Our use case is saving a small data in local cache and keep it consistent in every instances so that we can read all data in local cache without touching the database. Write will happens on both local cache and database. And database is only read on initialization. There is no read to database when the cluster is running.

**Reply by tristantarrant** [reply]
"Strong consistency" was used rather liberally there. Like most in-memory data grids, we are PA/EC. See https://en.wikipedia.org/wiki/PACELC_theorem.
Infinispan has an invalidation cache which might fit your needs: https://infinispan.org/docs/stable/titles/configuring/configuring.html#invalidation-caches_clustered-caches
This mode is what is used by our clustered Hibernate 2LC.
